tests/data/utils.py

```diff
@@ -14,5 +17,25 @@ def iterate_over_rules():
     """
     for dir_name, directories, files in os.walk(_DIR):
         leaf_dir = os.path.basename(dir_name)
+        rel_dir = '.' + dir_name[len(_DIR):]
         if leaf_dir.startswith('rule_'):
-            yield dir_name, _SSG_PREFIX + leaf_dir, files
+            yield rel_dir, _SSG_PREFIX + leaf_dir, files
+
+def _exclude_utils(file_name):
+    if file_name in ['./__init__.py', './utils.py']:
+        return True
+    if file_name.endswith('pyc'):
+        return True
+    if file_name.endswith('swp'):
+        return True
+    False
+
+def create_tarball(tar_dir):
+    archive_path = os.path.join(tar_dir, _TAR_BASENAME)
+    tarball = tarfile.TarFile.open(archive_path, 'w')
```

Comment:
use with tarfile.TarFile.open(...) as tarball, it's a nicer and safer pattern because any exception thrown will close the archive and cleanup

Reply:
*facepalm* yeah, I haven't realized I can use it. Thanks.